diff --git a/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.info b/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.info
index 5a180055..b53eac4f 100644
--- a/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.info
+++ b/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.info
@@ -7,6 +7,8 @@ dependencies[] = ckeditor
 
 stylesheets[all][] = css/ckeditor-accordion.css
 scripts[] = js/ckeditor-accordion.js
+configure = admin/settings/ckeditor_accordion
+
 ; Information added by Drupal.org packaging script on 2017-03-13
 version = "7.x-1.0"
 core = "7.x"
diff --git a/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.module b/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.module
index 595016a9..17efc059 100755
--- a/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.module
+++ b/sites/all/modules/contrib/ckeditor_accordion/ckeditor_accordion.module
@@ -1,25 +1,79 @@
 <?php
 
+/**
+ * @file
+ * Adds accordion plugin to CKEditor.
+ */
+
 /**
  * Implements hook_ckeditor_plugin().
- * Hook to register the CKEditor plugin - it would appear in the plugins list on the profile setting page.
+ *
+ * Hook to register the CKEditor plugin,
+ * it would appear in the plugins list on the profile setting page.
  */
 function ckeditor_accordion_ckeditor_plugin() {
   return array(
     'accordion' => array(
       // Name of the plugin used to write it.
       'name' => 'accordion',
-      // Description of the plugin - it would be displayed in the plugins management section of profile settings.
+      // Description of the plugin - it would be displayed in
+      // the plugins management section of profile settings.
       'desc' => t('CKEditor Accordion - A plugin to easily create accordions'),
-      // The full URL to the CKEditor plugins directory, with the trailing slash.
+      // The full URL to the CKEditor plugins directory,
+      // with the trailing slash.
       'path' => base_path() . drupal_get_path('module', 'ckeditor_accordion') . '/plugins/accordion/',
       'buttons' => array(
         'Accordion' => array(
           // Path to an icon relative to the plugins folder.
           'icon' => 'icons/accordion.png',
           'label' => 'Accordion',
-        )
-      )
-    )
+        ),
+      ),
+    ),
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function ckeditor_accordion_menu() {
+
+  $items = array();
+
+  $items['admin/settings/ckeditor_accordion'] = array(
+    'title' => 'Administer CKEditor Accordion',
+    'description' => 'Adminster CKEditor Accordion configuration',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ckeditor_accordion_admin'),
+    'access arguments' => array('administer ckeditor'),
+    'type' => MENU_NORMAL_ITEM,
   );
-}
\ No newline at end of file
+
+  return $items;
+}
+
+/**
+ * Create settings page for the module.
+ */
+function ckeditor_accordion_admin() {
+  $form = array();
+
+  $form['ckeditor_accordion_collapsed'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Collapse all tabs by default'),
+    '#default_value' => variable_get('ckeditor_accordion_collapsed', 0),
+    '#description' => t("The maximum number of links to display in the block."),
+  );
+
+  return system_settings_form($form);
+}
+
+/**
+ * Implements hook_page_build().
+ */
+function ckeditor_accordion_page_build(&$page) {
+  $config = variable_get('ckeditor_accordion_collapsed', 0);
+  if ($config) {
+    drupal_add_js(array('ckeditor_accordion' => array('collapseAll' => 1)), array('type' => 'setting'));
+  }
+}
diff --git a/sites/all/modules/contrib/ckeditor_accordion/css/ckeditor-accordion.css b/sites/all/modules/contrib/ckeditor_accordion/css/ckeditor-accordion.css
index 999566c4..9594dd71 100755
--- a/sites/all/modules/contrib/ckeditor_accordion/css/ckeditor-accordion.css
+++ b/sites/all/modules/contrib/ckeditor_accordion/css/ckeditor-accordion.css
@@ -103,7 +103,3 @@
   margin: 0;
   will-change: height;
 }
-
-.ckeditor-accordion-container > dl dd:first-of-type {
-  display: block;
-}
diff --git a/sites/all/modules/contrib/ckeditor_accordion/js/ckeditor-accordion.js b/sites/all/modules/contrib/ckeditor_accordion/js/ckeditor-accordion.js
index d4b118ec..7d36f6f7 100755
--- a/sites/all/modules/contrib/ckeditor_accordion/js/ckeditor-accordion.js
+++ b/sites/all/modules/contrib/ckeditor_accordion/js/ckeditor-accordion.js
@@ -19,9 +19,11 @@
           }
 
           // The first one is the correct one.
-          $accordion.children('dt:first').addClass('active');
-          $accordion.children('dd:first').addClass('active');
-
+          if (!Drupal.settings.ckeditor_accordion) {
+            $accordion.children('dt:first').addClass('active');
+            $accordion.children('dd:first').addClass('active');
+            $accordion.children('dd:first').css('display', 'block');
+          }
           // Turn the accordion tabs to links so that the content is accessible & can be traversed using keyboard.
           $accordion.children('dt').each(function () {
             var $tab = $(this);
