Index: rate.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rate.module	(revision ede95fb6c95966c2b993572b01d68edba54ecb68)
+++ rate.module	(revision )
@@ -455,14 +455,13 @@
   if ($entity_type == 'node' && module_exists('translation')) {
     if (arg(0) == 'node' && arg(1) == $entity_id) {
       // We are on the node page. Use node_load since the node is in static cache.
-      $entity_id = node_load($entity_id)->tnid;
+      $node = node_load($entity_id);
+      $entity_id = $node->tnid ? $node->tnid : $node->nid;
     }
     else {
       // We are not on the node page. Do not use node_load to prevent executing many useless queries.
-      $tnid = db_select('node', 'n')->fields('n', array('tnid'))->condition('n.nid', $entity_id)->execute()->fetchField();
-      if ($tnid) {
-        $entity_id = $tnid;
-      }
+      $result = db_select('node', 'n')->fields('n', array('tnid, nid'))->condition('n.nid', $entity_id)->execute()->fetchAssoc();
+      $entity_id = $result['tnid'] ? $result['tnid'] : $result['nid'];
     }
   }
   return $entity_id;
